### PR TITLE
fix(lobby): block creation of matches with invalid player counts

### PR DESCRIPTION
**Pull request by RevolutionTech**
#### Checklist

- [x] Use a separate branch in your local repo (not `main`).
- [x] Test coverage is 100% (or you have a story for why it's ok).

## Description

Fixes https://github.com/boardgameio/boardgame.io/issues/1055. Prevents a user from creating a match with an invalid number of players (only applicable if `minPlayers` and/or `maxPlayers` are defined).

## Testing

I tried to imitate the style and verbosity of the existing tests in `api.test.ts` with the tests I introduced, ~but technically some edge cases are not addressed by tests (eg. player count too low). If verbosity is preferred I'd be happy to add a few more test cases.~ but feel free to let me know if there's anywhere I can make improvements.

### CODE REVIEW

File: src/server/api.ts [hunk 1]
```diff
@@ -121,6 +121,14 @@ export const configureRouter = ({
     const game = games.find((g) => g.name === gameName);
     if (!game) ctx.throw(404, 'Game ' + gameName + ' not found');
 
+    if (
+      numPlayers != null &&
+      ((game.minPlayers != null && numPlayers < game.minPlayers) ||
+        (game.maxPlayers != null && numPlayers > game.maxPlayers))
```
Comment (delucis) [inline src/server/api.ts:127]: Could we avoid the loose inequality operator here? I find it can be hard to reason about all the different ways operands are compared.

I'd propose treating `numPlayers` more carefully because it's user input. We don't need to be so careful about the game config so I'd just check `minPlayers`/`maxPlayers` is truthy. Perhaps something like:

```js
    if (
      numPlayers !== undefined &&
        (typeof numPlayers !== 'number' ||
          (game.minPlayers && numPlayers < game.minPlayers) ||
          (game.maxPlayers && numPlayers > game.maxPlayers))
    ) {
```

That way we also throw an error if `numPlayers` is not actually a number.

**Review by delucis — CHANGES_REQUESTED** [review]
Thanks for this @RevolutionTech!

I've left a comment regarding the logic checking the validity of the input. More tests are always helpful especially for catching regressions later, but I'll leave it up to you how important it is to cover all edge cases.

File: src/server/api.ts (hunk 1 above)
Comment (RevolutionTech) [reply src/server/api.ts]: Sounds good. Since `Number.parseInt()` is used in the definition of `numPlayers` above and will always be a `number` (but could be `NaN`) I ended up using:
```javascript
ctx.request.body.numPlayers !== undefined && (Number.isNaN(numPlayers || ...
```

**Review by delucis — APPROVED** [review]
Nice catch with the `isNan`. Looks good to me — thanks!
